tokio/src/sync/mpsc/bounded.rs

```diff
@@ -155,6 +155,38 @@ impl<T> Receiver<T> {
         self.chan.recv(cx)
     }
 
+    /// Blocking receive to call outside of asynchronous contexts
+    ///
+    /// ```
+    /// use std::thread;
+    /// use tokio::runtime::Runtime;
+    /// use tokio::sync::mpsc;
+    ///
+    /// fn main() {
+    ///     let (mut tx, mut rx) = mpsc::channel::<u8>(10);
+    ///
+    ///     let sync_code = thread::spawn(move || {
+    ///         assert_eq!(Some(10), rx.blocking_recv());
+    ///     });
+    ///
+    ///     Runtime::new()
+    ///         .unwrap()
+    ///         .block_on(async move {
+    ///             let _ = tx.send(10).await;
+    ///         });
+    ///     sync_code.join().unwrap()
+    /// }
+    /// ```
+    #[cfg(any(feature = "blocking", feature = "rt-core", feature = "sync"))]
```

Comment:
The bounded channel is defined in a module that requires the `sync` feature, so `any(other_stuff, feature = "sync")` should always be satisifed?

Reply:
I tried without "sync" thinking that and it resulted in some unused item warnings in other modules that made the features CI job fail. The only way I could get rid of those warnings was by adding it

Reply:
Was that before you added `feature = "sync"` to `cfg_block_on`? I think you should give it another try now.

Reply:
That was the last failing commit 08197ebe9, so sync was in cfg_block_on

Reply:
Oh, the cfgs here is not what the error is about. It's due to warnings about things not being in use when only `sync` is enabled. Try removing the `feature = "sync"` clause from `cfg_block_on` and then change this function to
```rust
#[cfg(feature = "rt-core")]
#[cfg_attr(docsrs, doc(cfg(feature = "rt-core")))]
pub fn blocking_recv(&mut self) -> Option<T> {
```
I'm pretty sure this should work. Adding the other things from `cfg_block_on` is redundant, because if you look in [our `Cargo.toml`](https://github.com/tokio-rs/tokio/blob/1167c09ae8bfd0d4fff97d33803d959222f9e2ab/tokio/Cargo.toml#L50-L91), every other feature in the list requires `rt-core`.